shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -65,21 +73,14 @@ final class TsTypegenCodeBuilder {
     }
 
     // create a context with pre-created type var to name mapping
-    def apply(mlType: Type, typeVarMapping: MutMap[TypeVar, SourceCode]): TypegenContext = {
+    def apply(mlType: Type, typeVarMapping: MutSortedMap[TypeVar, SourceCode], isMethodDefintion: Boolean): TypegenContext = {
       val typegenTypeScope = Scope("localTypeScope", List.empty, typeScope)
       val typegenTermScope = Scope("localTermScope", List.empty, termScope)
-      TypegenContext(typeVarMapping, typegenTermScope, typegenTypeScope, false)
+      TypegenContext(typeVarMapping, typegenTermScope, typegenTypeScope, isMethodDefintion)
     }
   }
 
-  def declareTypeDef(typeDef: TypeDef): TypeSymbol = typeDef match {
-    case TypeDef(Als, TypeName(name), tparams, body, _, _) =>
-      typeScope.declareTypeAlias(name, tparams map { _.name }, body)
-    case TypeDef(Trt, TypeName(name), tparams, body, _, _) =>
-      typeScope.declareTrait(name, tparams map { _.name }, body)
-    case TypeDef(Cls, TypeName(name), tparams, baseType, _, members) =>
-      typeScope.declareClass(name, tparams map { _.name }, baseType, members)
-  }
+  def declareTypeDef(typeDef: TypeDef): TypeSymbol = typeScope.declareTypeSymbol(typeDef)
```

Comment:
Is this necessary? You're only using the method once, and its body is trivial.

Reply:
I'm doing this because typeScope is private variable. Is there some other way to do this?

Reply:
Ah right. I guess it's fine.